packages/medusa/src/services/cart.ts

```diff
@@ -864,17 +866,13 @@ class CartService extends TransactionBaseService {
           let currentItem: LineItem | undefined
 
           const existingItem = existingItemsVariantMap.get(item.variant_id)
-          if (item.should_merge) {
-            if (existingItem && isEqual(existingItem.metadata, item.metadata)) {
-              currentItem = existingItem
-            }
+          if (item.should_merge && existingItem) {
```

Comment:
question: We only merge if items are exactly 1:1. Having different metadata breaks this rule, so I think we need to keep the check.

Reply:
I discussed this with Adrien and we figured the check existed from before we got a proper `setMetadata` implementation, we also couldn't think of a case where this check made sense now that we can merge metadata, but I will be happy to hear if you know of such a case 😄 

Instead of having metadata dictate if we allow line-items to merge I added the `should_merge` property to the api for creating line_items since that explicitly controls if merges can occur.

Reply:
I am Christmas-shopping online and find an item that I want to buy as a gift. But I would also like to purchase it for myself because it is that good. The one I gift should be gift-wrapped.

I believe a simple way to differentiate between these two would be to use metadata. Something along the lines of:

```
let payload = {
  variant_id: variantId
  quantity: 1,
}

if (isGift) {
  payload = {
    ...payload,
    metadata: { gift_wrapping: true },
  }
}

await createLineItem(payload)
```

We can then react to this metadata field accordingly on the back. For example, to notify the warehouse, that something needs wrapping before it's shipped.

Reply:
alright, In the interest of time I will drop this change from here and refactor into the older update-line-item flow